src/psyGen.py

```diff
@@ -1272,14 +1537,40 @@ class HaloExchange(Node):
 
     def __init__(self, field, halo_type, halo_depth, check_dirty, parent=None):
         Node.__init__(self, children=[], parent=parent)
-        self._field = field
+        import copy
+        self._field = copy.copy(field)
```

Comment:
I was initially concerned that `copy.copy(None)` might not work but I've just checked and it is OK.